cmd/root_test.go

```diff
@@ -1536,6 +1536,232 @@ func TestGit(t *testing.T) {
 	)
 }
 
+func TestJujutsu(t *testing.T) {
+	as := require.New(t)
+
+	tempDir := test.TempExamples(t)
+	configPath := filepath.Join(tempDir, "/treefmt.toml")
+
+	test.ChangeWorkDir(t, tempDir)
+
+	// basic config
+	cfg := &config.Config{
+		FormatterConfigs: map[string]*config.Formatter{
+			"echo": {
+				Command:  "echo", // will not generate any underlying changes in the file
+				Includes: []string{"*"},
+			},
+		},
+	}
+
+	test.WriteConfig(t, configPath, cfg)
+
+	// init a jujutsu repo (disable signing for testing)
+	jjCmd := exec.Command("jj", "git", "init", "--config", "signing.backend=none")
+	as.NoError(jjCmd.Run(), "failed to init jujutsu repository")
+
+	// run before adding anything to the index
+	// we shouldn't pick up untracked files, because here the jujutsu walker differs from the git walker
```

Comment:
I have no knowledge of jj. I can't tell if this comment is telling me something fundamentally different about jj, or if it's telling me something about treefmt's jj walker implementation. Could you rephrase this to make it clear?